File: 01-Sorting/2-Selection Sort/selection_sort.py
def selection_sort(lst):
    length_list = len(lst)
    for i in range(length_list):
        min_val   = lst[i]
        min_index = i
  	# In each pass, the largest elements are pushed to the end
        for j  in range(i + 1, length_list):
            if lst[j] < min_val:
                min_val       = lst[j]
                min_index = j
    			#print(smallest)
        lst[i], lst[min_index] = lst[min_index], lst[i]
             
if __name__ == "__main__":
	lst = [5, -6, 8, 1, 4, 2, 0]
    # sort function
	selection_sort(lst)
	print(lst);

File: 01-Sorting/2-Selection Sort/selection_sort.c

//1- find smallest item of array
//2- Replace smallest item first one on array
//3- Expand sorted field
//4- Cotinue all untill all items are sorted.

#include <stdio.h>
#include <stdlib.h>

void selection_sort(int lst[], int n)
{
  int min = 0;
  int temp ;
  for (size_t i = 0; i < n; i++)
  {
    /* code */
    min = i;
    for (size_t j = i + 1; j < n; j++)
    {
        // if selected one bigger than other part
        if (lst[min] > lst[j])
        {
          min = j;
        }
     } 
     //replace
     temp = lst[i];
     lst[i] = lst[min];
     lst[min] = temp;
     
  }  

}


int main() 
{
  printf("hello world  -- selection-sort! \n");

  int lst[7] = {5, -6, 8, 1, 4, 2, 0};
  int n = 7;
  printf("before sort\n");
  for (size_t i = 0; i < n; i++)
  {
    printf("%4d ",lst[i]);
  }
  printf("\nafter sort\n");
  selection_sort(lst, n);
  for (size_t i = 0; i < n; i++)
  {
    printf("%4d ",lst[i]);
  }
  

  return 0; 

}
